### PR TITLE
Added the convert_ prefix in the ChartBlock

### PR DESCRIPTION
**Pull request by B3nedikt**

### CODE REVIEW

**Comment by B3nedikt** [comment]
To be honest I don't completely understand this module system xD But without this change it does not work for me locally. I would fix the tests if you think this is a good change ;)

Does it just take the module name and prefixe it with `convert_` ?

**Comment by ggrossetie** [comment]
> But without this change it does not work for me locally.

@B3nedikt could you please open an issue with a simple reproduction case please? Thanks!

> Does it just take the module name and prefixe it with convert_ ?

Yes the converter will append `convert_`. You can see that the tests are failing:

```
     NoMethodError:
       undefined method `convert_convert_chart' for #<Asciidoctor::Converter::Html5Converter:0x000055dc48cafdd0>
       Did you mean?  convert_chart
```

**Comment by ggrossetie** [comment]
It seems that the new converters should use prefixed method names according to this comment:

https://github.com/asciidoctor/asciidoctor/blob/4bab183b9f3fad538f86071b2106f3b8185d3832/lib/asciidoctor/converter/html5.rb#L1347-L1350

So maybe we need to fix something but without a reproduction case I can only guess.

File: lib/asciidoctor/chart/chart_block.rb [hunk 1]
```diff
@@ -13,7 +13,7 @@ def initialize(parent, data, attrs)
                                          'engine' => engine,
                                          'data-raw' => data
                                        })
-        super parent, :chart, { source: nil, attributes: block_attributes, subs: nil }
```
Comment (ggrossetie) [inline lib/asciidoctor/chart/chart_block.rb:16]: `:chart` is the context name not the actual name of the convert method. So I would say that it should be `:chart`.

Comment (B3nedikt) [reply lib/asciidoctor/chart/chart_block.rb:16]: Yeah, it definitely should be. I think the issue I am facing is specific to my setup.

I have the following Gemfile:

```ruby
source 'https://rubygems.org'

gem 'asciidoctor-revealjs', '5.0.0.rc1'

gem 'asciidoctor-chart', '1.0.0'

gem 'tilt', '~> 2.0.0'
``` 

Using ruby 3.1.2.

And I get the following error:
```
 `method_missing': undefined method `chart' for #<Asciidoctor::Converter::Html5Converter
```

If I change it as in this PR, the error goes away, and everything works as expected. Which it should not as you also described above xD

Unfortunately I can't just link my project, as it is for internal use. I close this PR for now, and try to properly reproduce it another time. I think it is likely just some dependency incompatibility case...

Comment (ggrossetie) [reply lib/asciidoctor/chart/chart_block.rb:16]: This is actually a bug/limitation in Asciidoctor reveal.js. The converter should support prefixed method `convert_`

https://github.com/asciidoctor/asciidoctor-reveal.js/blob/19cfd091abd1d5151b614784218cf0feadc6bdef/lib/asciidoctor-revealjs/converter.rb#L399-L413

I will open a new issue.

Comment (jschwartzenberg) [reply lib/asciidoctor/chart/chart_block.rb:16]: @ggrossetie did you still open a new issue? I'm running into the exact same issue. At least I found a workaround here :)

Comment (ggrossetie) [reply lib/asciidoctor/chart/chart_block.rb:16]: It was fixed in https://github.com/jirutka/asciidoctor-templates-compiler/commit/f4eee46264de38d41278ce45fe8641c9f301101a but there's no release yet. @jschwartzenberg please request a new release at https://github.com/jirutka/asciidoctor-templates-compiler/issues

Comment (jirutka) [reply lib/asciidoctor/chart/chart_block.rb:16]: I released 0.7.0. Sorry about the delay, I forgot about it before.

Comment (ggrossetie) [reply lib/asciidoctor/chart/chart_block.rb:16]: Yay, thanks Jakub!

Comment (B3nedikt) [reply lib/asciidoctor/chart/chart_block.rb:16]: Thanks :)

Comment (jschwartzenberg) [reply lib/asciidoctor/chart/chart_block.rb:16]: I'm still seeing this issue. Maybe I am missing something I should change in my `Gemfile`? I saw Asciidoctor reveal.js has release 5.0.1 now.

Comment (ggrossetie) [reply lib/asciidoctor/chart/chart_block.rb:16]: Not yet merged: https://github.com/asciidoctor/asciidoctor-reveal.js/pull/515
I will be part of Asciidoctor reveal.js 5.1.0.

**Pull request closed without merging**